Source/JavaScriptCore/runtime/TemporalPlainTime.cpp

```diff
@@ -526,52 +527,49 @@ static ISO8601::Duration differenceTime(ISO8601::PlainTime time1, ISO8601::Plain
+    JSGlobalObject* globalObject, TemporalPlainTime* other, JSValue optionsValue) const
 {
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
     auto [smallestUnit, largestUnit, roundingMode, increment] = extractDifferenceOptions(globalObject, optionsValue, UnitGroup::Time, TemporalUnit::Nanosecond, TemporalUnit::Hour);
     RETURN_IF_EXCEPTION(scope, { });
 
-    auto result = differenceTime(plainTime(), other->plainTime());
-    result.setYears(0);
-    result.setMonths(0);
-    result.setWeeks(0);
-    result.setDays(0);
-    TemporalDuration::round(result, increment, smallestUnit, roundingMode);
-    TemporalDuration::balance(result, largestUnit);
-    return result;
+    Int128 timeDuration;
+
+    // The sign of the argument to roundTimeDuration() determines the choice
+    // of rounding mode, so the sign should be preserved here instead of
+    // negating the duration at the end
+    if (operation == DifferenceOperation::Since)
+        timeDuration = differenceTime(other->plainTime(), plainTime());
+    else
+        timeDuration = differenceTime(plainTime(), other->plainTime());
+
+    auto d = ISO8601::roundTimeDuration(globalObject, timeDuration, increment, smallestUnit, roundingMode);
+    RETURN_IF_EXCEPTION(scope, { });
+    auto duration = ISO8601::InternalDuration::combineDateAndTimeDuration(ISO8601::Duration(), d);
+    return TemporalDuration::temporalDurationFromInternal(duration, largestUnit);
 }
 
-ISO8601::Duration TemporalPlainTime::since(JSGlobalObject* globalObject, TemporalPlainTime* other, JSValue optionsValue) const
+// https://tc39.es/proposal-temporal/#sec-temporal.plaintime.prototype.until
+ISO8601::Duration TemporalPlainTime::until(JSGlobalObject* globalObject, TemporalPlainTime* other, JSValue optionsValue) const
 {
-    VM& vm = globalObject->vm();
-    auto scope = DECLARE_THROW_SCOPE(vm);
+    return differenceTemporalPlainTime(DifferenceOperation::Until,
+        globalObject, other, optionsValue);
+}
 
-    auto [smallestUnit, largestUnit, roundingMode, increment] = extractDifferenceOptions(globalObject, optionsValue, UnitGroup::Time, TemporalUnit::Nanosecond, TemporalUnit::Hour);
-    RETURN_IF_EXCEPTION(scope, { });
-    roundingMode = negateTemporalRoundingMode(roundingMode);
-
-    auto result = differenceTime(other->plainTime(), plainTime());
-    result = -result;
-    result.setYears(0);
-    result.setMonths(0);
-    result.setWeeks(0);
-    result.setDays(0);
-    TemporalDuration::round(result, increment, smallestUnit, roundingMode);
-    result = -result;
-    result.setDays(0);
-    TemporalDuration::balance(result, largestUnit);
-    return result;
+// https://tc39.es/proposal-temporal/#sec-temporal.plaintime.prototype.since
+ISO8601::Duration TemporalPlainTime::since(JSGlobalObject* globalObject, TemporalPlainTime* other, JSValue optionsValue) const
+{
+    return differenceTemporalPlainTime(DifferenceOperation::Since,
+        globalObject, other, optionsValue);
```

Comment:
Ditto.

Reply:
Fixed in [66fe9a7](https://github.com/WebKit/WebKit/pull/50388/commits/66fe9a740f5b5c2e6e103fb24c5924d3fa2b8735)